### PR TITLE
Unfinished pull

### PR DESCRIPTION
**Pull request by Fl4m3Ph03n1x**
Here. 

I was super original and named it "unfinished pull". 
It passes all the tape tests, for what it's worth. 
The bench script also runs. 
Updated several vulnerable dependencies. 
No more security issues with Buffer. 
Needs Node v6 or higher.

I could use a hand in testing it with a real system though, as I didn't have enough time for it yet, but all seems to be well.

The merge was quite hard to be honest. But anyway, what do you think?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,19 +21,27 @@
...
   "scripts": {
-    "test": "node ./test"
+    "test": "tape ./test/*.js | tap-spec",
+    "cover": "nyc tape ./test/*.js",
+    "lint": "jshint ./lib",
+    "bench": "node ./bench.js"
+  },
+  "dependencies": {
+    "jshint": "^2.9.5",
+    "nyc": "^12.0.2"
```
Comment (ronkorving) [inline package.json]: These are dev deps :)

File: lib/Graylog.js [hunk 2]
```diff
@@ -424,4 +424,3 @@ Graylog.prototype.close = function (cb) {
 
 
 module.exports = Graylog;
-module.exports.graylog = Graylog; // deprecated
```
Comment (ronkorving) [inline lib/Graylog.js]: This one line keeps API compatibility between the old and the new. That's why I added it to the re-implementation.

File: package.json (hunk 1 above)
Comment (Fl4m3Ph03n1x) [reply package.json]: Yes they are, I wonder what they are doing there. 
I will fix that and try again!

File: lib/Graylog.js (hunk 2 above)
Comment (Fl4m3Ph03n1x) [reply lib/Graylog.js]: Is it really necessary though? 
I did have the idea that this was supposed to be a major version bump, which means it won't be backwards compatible.

**Comment by Fl4m3Ph03n1x** [comment]
moved dependencies to DEV, updated changelog with some preliminar information ( will need to confirm and change as we see what has actually be done and what has not ). 

For me, personally, I would make the log events async ( via Promises ) and get rid of the need to close the Graylog object for it to empty the queue. This need to close the socket is also not documented at all and I argue that we should ditch it altogether and just make the logs async. 

Since this is supposed to be a major version bump, why not get rid of all the issues we currently have?

**Comment by Fl4m3Ph03n1x** [comment]
Hello there, do you have any news on this? I would like to have some more feedback to know what I can change so this can be approved.

**Comment by ronkorving** [comment]
A lot of indentation got wonky. Please have a look at `Graylog.js`.

**Pull request closed without merging**